src/main/proto/marathon.proto

```diff
@@ -148,11 +149,42 @@ message ExtendedContainerInfo {
   }
 
   required mesos.ContainerInfo.Type type = 1;
-  repeated mesos.Volume volumes = 2;
+  repeated Volume volumes = 2;
 
   optional DockerInfo docker = 3;
 }
 
+// Copied from mesos.proto and adjusted to meet the needs of persistent volumes
+// Describes a volume mapping either from host to container or vice
+// versa. Both paths can either refer to a directory or a file.
+message Volume {
+  required mesos.Volume.Mode mode = 3;
+
+  // Path pointing to a directory or file in the container. If the
+  // path is a relative path, it is relative to the container work
+  // directory. If the path is an absolute path, that path must
+  // already exist.
+  required string container_path = 1;
+
+  // The following specifies the source of this volume. At most one of
+  // the following should be set.
+
+  // Absolute path pointing to a directory or file on the host or a
+  // path relative to the container work directory.
+  optional string host_path = 2;
+
+  // The source of the volume is an Image which describes a root
+  // filesystem which will be provisioned by Mesos.
+  optional mesos.Image image = 4;
+
+  // Defines a Persistent Volume. In this case hostPath must not be set
+  optional PersistentVolumeInfo persistent = 5;
+}
+
+message PersistentVolumeInfo {
```

Comment:
Can you define that message inside Volume?
So you have all Volume related type definitions together.